drivers/pwm/pwm_nrfx.c

```diff
@@ -243,12 +249,17 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 static int pwm_nrfx_get_cycles_per_sec(const struct device *dev, uint32_t channel,
 				       uint64_t *cycles)
 {
-	/* TODO: Since this function might be removed, we will always return
-	 * 16MHz from this function and handle the conversion with prescaler,
-	 * etc, in the pin set function. See issue #6958.
-	 */
+#if defined(NRF_PWM120)
+	const struct pwm_nrfx_config *config = dev->config;
+	if(config->pwm.p_reg == NRF_PWM120)
+	{
+		*cycles = 320ul * 1000ul * 1000ul;
+		
+		return 0;
+	}
+#endif
```

Comment:
I am affraid that would require adding clock property to every nrf pwm node. If thats not an issue, then sure.

Reply:
all that need it, yes. Just check the UART case.

Reply:
it can be made optional, see

https://github.com/zephyrproject-rtos/zephyr/blob/124aae3876842154964726ba6e89565040648f9a/drivers/serial/uart_nrfx_uarte.c#L2290-L2294